web/libs/datamanager/src/stores/Tabs/store.js

```diff
@@ -333,6 +333,23 @@ export const TabStore = types
       return view;
     }),
 
+    updateViewOrder: flow(function* (source, destination) {
+      // Detach the view from the original position
+      const [removed] = self.views.splice(source, 1);
+      const sn = getSnapshot(removed);
+
+      // Insert the view at the new position
+      self.views.splice(destination, 0, sn);
```

Comment:
Probably these manipulations are the reason for errors on tab duplication.
`self.views` is changed in place, so no re-renders happen. So rendered tabs are the same old objects even after updates, because original tab is removed and new one is created from snapshot. So rendered tab doesn't have parent anymore.
To fix this you can first try to update `views` in immutable manner by updating it: `self.views = [...self.views]`. Then all rendered tabs will be the actual ones, but this can slow tabs down. You can also try to change the way you access parent, basically it's just `views`.

Reply:
Yeah, I started using views instead tab.parent, it's easier and it doesn't depend of any reference, I believe that in this case this solution is better. Thanks @hlomzik